R/qenv-eval_code.R

```diff
@@ -28,24 +28,45 @@ setMethod("eval_code", signature = c("qenv", "expression"), function(object, cod
   object@env <- rlang::env_clone(object@env, parent = parent.env(.GlobalEnv))
   object@code <- c(object@code, code)
 
-  tryCatch(
-    {
-      eval(code, envir = object@env)
-      lockEnvironment(object@env, bindings = TRUE)
-      object
+  current_warnings <- ""
+  current_messages <- ""
+
+  x <- withCallingHandlers(
+    tryCatch(
+      {
+        eval(code, envir = object@env)
+        lockEnvironment(object@env, bindings = TRUE)
+        NULL
+      },
+      error = function(e) {
+        errorCondition(
+          message = sprintf(
+            "%s \n when evaluating qenv code:\n %s",
+            conditionMessage(e),
+            paste(code, collapse = "\n ")
+          ),
+          class = c("qenv.error", "try-error", "simpleError"),
+          trace = object@code
+        )
+      }
+    ),
+    warning = function(w) {
+      current_warnings <<- paste(current_warnings, suppressWarnings(as.character(w)))
```

Comment:
As we have slots for warnings and messages we may be able to replace:

`current_warnings <- ""` with `object@warnings <- c(object@warnings, "")` and then

`current_warnings <<- paste(current_warnings, suppressWarnings(as.character(w)))` with

`object@warnings[length(object@warnings)] <- paste(object@warnings[length(object@warnings)] , suppressWarnings(...)`

and then remove lines 66 and 67? As we have tests we can easily check

Reply:
We go back to the issue where object@warnings  keeps re initializing and rendering empty warnings slot :(